test_case/ERA5TimeSeriesDataset.py

```diff
@@ -88,6 +91,21 @@ def __getitem__(self, index):
         x_data = self.ds_x.isel(time=index).to_array().values
         y_data = self.ds_y.isel(time=index).to_array().values
         return (x_data, y_data)
+
+    def __iter__(self):
+        """Reset index for new iteration"""
+        self.index = 0
+        return self
+
+    def __next__(self):
+        """Return (input, target) pair with proper forecast offset"""
+        if self.index < self.length:
+            # Get input and target at current index
+            x_data = self.ds_x.isel(time=self.index).to_array().values
+            y_data = self.ds_y.isel(time=self.index).to_array().values
+            self.index += 1
+            return (x_data, y_data)
+        raise StopIteration
```

Comment:
Raise StopIteration only when `self.index >= self.length`?

```suggestion
        else:
            raise StopIteration
```